src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -0,0 +1,695 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// Represents a data structure in which each element additionally has a priority
+    /// associated with it.
+    /// </summary>
+    /// <typeparam name="TElement">The type of the element.</typeparam>
+    /// <typeparam name="TPriority">The type of the priority.</typeparam>
+    public class PriorityQueue<TElement, TPriority>
+    {
+        /// <summary>
+        /// Represents an implicit heap-ordered complete d-ary tree, stored as an array.
+        /// </summary>
+        private (TElement element, TPriority priority)[] _nodes;
```

Comment:
As such tuples are stand-ins for custom struct types, and as the members here would be public on that struct and named accordingly, we prefer such tuples to instead be:
```C#
private (TElement Element, TPriority Priority)[] _nodes;
```
That way, consuming code looks like:
```C#
return root.Element;
```
as it would if this were a custom type, instead of:
```C#
return root.element;
```

Reply:
Could you elaborate on why this is desired?

Reply:
My first sentence above:
"As such tuples are stand-ins for custom struct types, and as the members here would be public on that struct and named accordingly"
Happy to elaborate,  but what specifically needs more elaboration? Our style is that public members are PascalCased, and these are effectively public members.